convert_hf_to_gguf.py

```diff
@@ -81,11 +81,25 @@ class ModelBase:
     block_count: int
     tensor_map: gguf.TensorNameMap
 
-    def __init__(self, dir_model: Path, ftype: gguf.LlamaFileType, fname_out: Path, *, is_big_endian: bool = False,
-                 use_temp_file: bool = False, eager: bool = False,
-                 metadata_override: Path | None = None, model_name: str | None = None,
-                 split_max_tensors: int = 0, split_max_size: int = 0, dry_run: bool = False,
-                 small_first_shard: bool = False, hparams: dict[str, Any] | None = None, remote_hf_model_id: str | None = None):
+    def __init__(
+        self,
+        dir_model          : Path,
+        ftype              : gguf.LlamaFileType,
+        fname_out          : Path,
+        hf_arch            : str,
```

Comment:
btw, if we update it here, we should also update `convert_lora_to_gguf`, that's why I think it's better not to add to many input arguments for this class. `hf_arch` can indeed be implied from `hparams`, so could we remove it from this list?